detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/BracesOnIfStatements.kt

```diff
@@ -0,0 +1,226 @@
+
+/**
+ * This rule detects `if` statements which do not comply with the specified rules.
+ * Keeping braces consistent would improve readability and avoid possible errors.
+ *
+ * The available options are:
+ *  * `always`: forces braces on all `if` and `else` branches in the whole codebase.
+ *  * `consistent`: ensures that braces are consistent within each `if`-`else if`-`else` chain.
+ *    If there's a brace on one of the branches, all branches should have it.
+ *  * `necessary`: forces no braces on any `if` and `else` branches in the whole codebase
+ *    except where necessary for multi-statement branches.
+ *  * `never`: forces no braces on any `if` and `else` branches in the whole codebase.
+ *
+ * SingleLine if-statement has no line break (\n):
+ * ```
+ * if (a) b else c
+ * ```
+ * MultiLine if-statement has at least one line break (\n):
+ * ```
+ * if (a) b
+ * else c
+ * ```
+ *
+ * <noncompliant>
+ *
+ * // singleLine = 'never'
+ * if (a) { b } else { c }
+ *
+ * if (a) { b } else c
+ *
+ * if (a) b else { c; d }
+ *
+ * // multiLine = 'never'
+ * if (a) {
+ *    b
+ * } else {
+ *    c
+ * }
+ *
+ * // singleLine = 'always'
+ * if (a) b else c
+ *
+ * if (a) { b } else c
+ *
+ * // multiLine = 'always'
+ * if (a) {
+ *    b
+ * } else
+ *    c
+ *
+ * // singleLine = 'consistent'
+ * if (a) b else { c }
+ * if (a) b else if (c) d else { e }
+ *
+ * // multiLine = 'consistent'
+ * if (a)
+ *    b
+ * else {
+ *    c
+ * }
+ * </noncompliant>
+ *
+ * <compliant>
+ * // singleLine = 'never'
+ * if (a) b else c
+ *
+ * // multiLine = 'never'
+ * if (a)
+ *    b
+ * else
+ *    c
+ *
+ * // singleLine = 'always'
+ * if (a) { b } else { c }
+ *
+ * if (a) { b } else if (c) { d }
+ *
+ * // multiLine = 'always'
+ * if (a) {
+ *    b
+ * } else {
+ *    c
+ * }
+ *
+ * if (a) {
+ *    b
+ * } else if (c) {
+ *    d
+ * }
+ *
+ * // singleLine = 'consistent'
+ * if (a) { b } else { c }
+ *
+ * if (a) b else c
+ *
+ * if (a) { b } else { c; d }
+ *
+ * // multiLine = 'consistent'
+ * if (a) {
+ *    b
+ * } else {
+ *    c
+ * }
+ *
+ * if (a) b
+ * else c
+ * </compliant>
+ */
+class BracesOnIfStatements(config: Config = Config.empty) : Rule(config) {
+
+    override val issue = Issue(
+        javaClass.simpleName,
+        Severity.Style,
+        "Braces do not comply with the specified policy",
+        Debt.FIVE_MINS
+    )
+
+    @Configuration("single-line braces policy")
+    private val singleLine: BracePolicy by config("never") { BracePolicy.getValue(it) }
+
+    @Configuration("multi-line braces policy")
+    private val multiLine: BracePolicy by config("always") { BracePolicy.getValue(it) }
+
+    override fun visitIfExpression(expression: KtIfExpression) {
+        super.visitIfExpression(expression)
+
+        val parent = expression.parent.parent
+        val parentCondition = (parent as? KtIfExpression)?.condition
+        if ((parent is KtIfExpression) && (parentCondition !== expression)) return
```

Comment:
This seems equivalent and simpler:
```suggestion
        // Ignore `then` and `else` branches, they're handled by the initial `if`'s visit.
        // But let us process conditions, because they might be `if` themselves.
        if (parent is KtIfExpression && parent.condition !== expression) return
```